### PR TITLE
New homepage  gov activity

### PR DESCRIPTION
**Pull request by jon-kirwan**
## What
https://trello.com/c/DhyxNleq/644-homepage-government-activity-big-numbers, [Jira issue NAV-3276](https://gov-uk.atlassian.net/browse/NAV-3276)

Build the government activity and big numbers section of the homepage based on the new design.

## Why
The design of the homepage hasn't been updated in quite a while - this updated design will help the homepage better meet the needs of GOV.UK's users.

## Visual changes
See https://govuk-fronte-new-homepa-d3fkpq.herokuapp.com/

<table role="table">
<thead>
<tr>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td valign="top"><a target="_blank" rel="noopener noreferrer" href="https://user-images.githubusercontent.com/87758239/143443359-19689f94-d472-4fd5-8d20-e6c86665c3c8.png"></a></td>
<td valign="top"><a target="_blank" rel="noopener noreferrer" href="https://user-images.githubusercontent.com/87758239/143441259-b5f714c2-8db1-4de0-a83a-69d837daab58.png"></a></td>
</tr>
<tr>
<td valign="top"><a target="_blank" rel="noopener noreferrer" href="https://user-images.githubusercontent.com/87758239/143443245-7de19e02-0703-4169-94ca-765912bb87bf.png"></a></td>
<td valign="top"><a target="_blank" rel="noopener noreferrer" href="[#](https://user-images.githubusercontent.com/87758239/143441270-e7fb6ad2-cc96-4f40-a81c-446178fe71f7.png)"></a></td>
</tr>
</tbody>
</table>

## Anything else
- @mia-allers-gds - i've had a bit of trouble getting the text 'Find out what the government is doing and how to get involved' to display on one line which affects the alignment with the 'Departments and organisations' content in the right column. I think it's because of the column gutters which mean the width is more like 630px rather than 660px(ish)
- ~~i've reused the chevron card pattern (styles, markup and mixins) from https://github.com/alphagov/frontend/pull/3033~~
- ~~I think we may need to add an extra yaml entry for one of the tracking dimensions. `track-action` is set to `topicsLink` for the topics but should be set to `governmentactivityLink` for the Government Activity cards~~
- i've also created a section and heading pattern (also used in https://github.com/alphagov/frontend/pull/3029) as it's not possible to achieve the same look with design system classes only.

CSS
```
.homepage-section {
  margin: govuk-spacing(4) 0 govuk-spacing(9);
}

.homepage-section__heading {
  border-top-style: solid;
  border-top-width: 2px;
  margin: 0 0 govuk-spacing(6);
  padding: govuk-spacing(4) 0 0;
}
```
HTML
```
<section class="homepage-section homepage-section__government_activity">
  <div class="govuk-grid-row home-numbers-and-info">
    <div class="govuk-grid-column-two-thirds" data-module="gem-track-click">
      <div class="homepage-section__heading">
        <%= render "govuk_publishing_components/components/heading", {
          text: t('homepage.index.government_activity'),
          margin_bottom: 2,
        } %>
```

### CODE REVIEW

**Review by owenatgov — DISMISSED** [review]
Requesting changes off of this one as well I'm afraid. These changes need to be branched off of the `new-homepage` branch, not main.

**Comment by injms** [comment]
I've updated the chevron card partial to be more flexible (https://github.com/alphagov/frontend/pull/3033/commits/f50d2b5ff133ae6f69b09d2dbbe3f120be999bd1) - which means that the action can be set to `governmentactivityLink` without needing to set it in the YAML file.

File: app/assets/stylesheets/views/_homepage.scss
Comment (injms) [inline app/assets/stylesheets/views/_homepage.scss:190]: It'd be good to follow the BEM convention of using hyphens in element name rather than underscores:

```suggestion
  .homepage-section__government-activity & {
```

File: app/assets/stylesheets/views/_homepage.scss
Comment (injms) [inline app/assets/stylesheets/views/_homepage.scss:275]: Overriding a component's CSS is brittle because it heavily depends on the component staying the same, which isn't something that's guaranteed. Is there another way we can achieve this spacing?

File: app/views/homepage/_government_activity.erb
Comment (injms) [inline app/views/homepage/_government_activity.erb:12]: Let's use double quotes here:
```suggestion
          text: t("homepage.index.government_activity"),
```

**Comment by mia-allers-gds** [comment]
@jon-kirwan I've noticed the spacing after the government activity section on mobile is wrong. It should be 50px

Could you change the text under government activity to read: "Find out what the government is doing"

File: app/views/homepage/_government_activity.erb
Comment (jon-kirwan) [reply app/views/homepage/_government_activity.erb]: Updated and for all other occurrences in the view.

File: app/assets/stylesheets/views/_homepage.scss
Comment (jon-kirwan) [reply app/assets/stylesheets/views/_homepage.scss:275]: I could add a wrapper div / class around the component and target that instead?

**Comment by jon-kirwan** [comment]
> Find out what the government is doing

Thanks, both changes added.

**Comment by mia-allers-gds** [comment]
@jon-kirwan I see the content change but not the extra vertical space after government activity (see screenshot)
Screenshot 2021-11-25 at 16 14 41

Spacing shown in spec:
Screenshot 2021-11-25 at 16 16 24

File: app/assets/stylesheets/views/_homepage.scss
Comment (injms) [reply app/assets/stylesheets/views/_homepage.scss:275]: I think that's a better option - and then could you add a issue on the components gem about adding the ability to set the margin on the component? 

Once that feature has been added we can then remove the wrapping element and add the spacing using the component - but that can be done at a later date.

Comment (jon-kirwan) [reply app/assets/stylesheets/views/_homepage.scss]: Done. Created an issue here: https://github.com/alphagov/govuk_publishing_components/issues/2479

**Comment by jon-kirwan** [comment]
> @jon-kirwan I see the content change but not the extra vertical space after government activity (see screenshot) Screenshot 2021-11-25 at 16 14 41
> 
> Spacing shown in spec: Screenshot 2021-11-25 at 16 16 24

@mia-allers-gds - apologies, missed that one. Should be sorted now

**Comment by mia-allers-gds** [comment]
@jon-kirwan thank you, looks good!

File: app/views/homepage/_government_activity.erb
Comment (owenatgov) [inline app/views/homepage/_government_activity.erb:1]: A tiny little nitpick that this should be `government_activity.html.erb` not just `.erb`

File: app/assets/stylesheets/views/_homepage.scss
Comment (owenatgov) [inline app/assets/stylesheets/views/_homepage.scss:177]: This is a very crafty solution but I wonder if it's very interpretable for devs. It does mean you're repeating yourself less by reusing those list reset styles but you're also effectively hiding the functionality inside a class that has a name tied to a different section. Maybe we could create generic list class to use between the sections but I'd be tempted to just repeat the CSS for this section and give it it's own class with a name that's explicit to that section. This is probably something for us to consider about if we can make this scss file a bit more efficient once we've deployed it.

I'm not gonna block this from going out if you don't agree as it's mostly a stylistic suggestion. I leave this to you. Happy to discuss.

File: app/views/homepage/_government_activity.erb
Comment (owenatgov) [inline app/views/homepage/_government_activity.erb:17]: This could be a [lead paragraph with a custom margin bottom of 0](https://components.publishing.service.gov.uk/component-guide/lead_paragraph/custom_margin_bottom)

File: config/locales/en.yml
Comment (owenatgov) [inline config/locales/en.yml:316]: Because we're doing some explicit column setting, I'd advise adding a comment reminding folks that if they add something here, they need to updated the associated css. [Here's an example on my more on govuk PR](https://github.com/alphagov/frontend/pull/3039/files#diff-44438ce218f5287c58d0017f965d888715635d94280669896f75841fbd7b4cd7R363-R364).

File: app/views/homepage/_government_activity.erb
Comment (owenatgov) [inline app/views/homepage/_government_activity.erb:46]: Nice 💪

File: app/views/homepage/_government_activity.erb
Comment (owenatgov) [inline app/views/homepage/_government_activity.erb:60]: A teeny weeny nitpick that this was originally `raw` because of the need for the text to go over 2 lines so it looks nice. I'm pretty confident that we don't need it anymore.

**Review by owenatgov — COMMENTED** [review]
Sadly, I have some opinions. We're very nearly there though!

**Comment by owenatgov** [comment]
Something else is that before we merge this, could you try to clear up the commit history so it's a bit more logical and verbose?

File: app/views/homepage/_government_activity.erb
Comment (jon-kirwan) [reply app/views/homepage/_government_activity.erb]: Nice one, completely forgot I could use this as a component.

**Comment by jon-kirwan** [comment]
> Something else is that before we merge this, could you try to clear up the commit history so it's a bit more logical and verbose?

Yep no worries, i've added the commits like this to (hopefully) make it easier to follow the updates. I'll squash them all down though once it's approved.

File: app/views/homepage/_government_activity.erb
Comment (jon-kirwan) [reply app/views/homepage/_government_activity.erb]: Yep that's good. Makes no difference when removed (I've also removed an `&nbsp;` from the locale entry too)

File: app/assets/stylesheets/views/_homepage.scss
Comment (jon-kirwan) [reply app/assets/stylesheets/views/_homepage.scss:177]: Thanks Owen, definitely understand the concerns here.

My thinking around this approach is that a parent selector requires less code to write but can help with readability as you can see in one block how the mixin can be modified based on the context - I think more comments will be needed though to explain that?

With 3 variants used across the home page it could be written something like this -

```
.homepage-generic-grid-name {
  @include columns($items: 16, $columns: 1, $selector: "li");
  list-style: none;
  margin: 0 govuk-spacing(-3);
  padding: 0;

  @include govuk-media-query($from: "tablet") {
    @include columns($items: 16, $columns: 2, $selector: "li");
  }

  @include govuk-media-query($from: "desktop") {
    @include columns($items: 16, $columns: 3, $selector: "li");
  }
  
  // Government activity
  .homepage-section__government-activity & {
    @include columns($items: 5, $columns: 1, $selector: "li");
    margin-bottom: govuk-spacing(4);

    @include govuk-media-query($from: "tablet") {
      @include columns($items: 5, $columns: 2, $selector: "li");
      margin-bottom: 0;
    }
  }
  
  // Most active
  .homepage-section__most-active & {
    @include govuk-media-query($from: desktop) {
      @include columns($items: 10, $columns: 2, $selector: "li", $flow: column);
    }
  }
}

I might leave things as they are for the moment... but definitely think we should revisit this when everything is pulled together in `new-homepage` and then decide on one approach 👍

Comment (owenatgov) [reply app/assets/stylesheets/views/_homepage.scss:177]: I can live with that. Are you ok to add this as a note to the [tech debt card](https://trello.com/c/RGlBCzuC/186-record-of-the-tech-debt-created-from-homepage-sprints-that-needs-clearing-up-post-deploy)?

Something else that occurs to me is that we could potentially filter all this grid stuff into the inevitable component so this ingests the grid values as well as the content to render this component. That would save all this grid chaff being needed here.

**Review by owenatgov — APPROVED** [review]
I'm happy 👍  If you could please tidy up the commit history before merging that'd be grand!

File: app/assets/stylesheets/views/_homepage.scss
Comment (jon-kirwan) [reply app/assets/stylesheets/views/_homepage.scss:177]: Yep no worries (and I was thinking the same thing about the grid values too after i'd posted this). I'll update all on the tech debt card.

**Pull request merged**
